docs/managing-crd-versions.md

```diff
@@ -0,0 +1,296 @@
+   - Existing values remain valid
+
+## Breaking vs. Non-Breaking Changes
+
+### Non-Breaking Changes (Backward Compatible)
+
+When new fields are **optional** or have **default values**, no version bump is
+necessary. The existing API version remains backward compatible because:
+
+- Old manifests that don't specify the new fields continue to work
+- The Terraform provider handles the absence of these fields gracefully
+- Late initialization can populate sensible defaults
+
+**Example**: Adding an optional `encryption_enabled` field to a database
+resource doesn't break existing manifests that don't specify this field.
+
+### Breaking Changes (Require Versioning)
+
+Breaking changes require careful management through versioning:
+
+#### Hiding Breaking API Changes
+
+When fields are renamed or removed, generating a new API version allows old
+manifests to continue working while new manifests can use the updated schema.
+
+**Example**: If a Terraform resource renames `instance_name` to `name`:
+- `v1alpha1`: Keep `instance_name` field (for existing manifests)
+- `v1alpha2`: Use new `name` field (for new manifests)
+- Conversion webhook: Translate between the two versions
+
+The new version contains updated field names while the old version preserves
+original field names, "shielding clients of the old API version from the
+breaking change."
+
+#### Unavoidable Breaking Changes
+
+Some breaking changes cannot be fully mitigated by versioning:
+
+**Required Fields**: When a new field becomes required (not optional):
+- Both old and new API versions will fail to create resources without this field
+- This represents an unavoidable breaking change at the provider level
+- Users must update their manifests regardless of the API version they use
+
+**Recommendation**: Communicate these changes clearly in release notes and
+migration guides.
+
+## Implementation with Upjet
+
+Upjet provides a resource configuration API to manage multiple CRD versions:
+
+### Configuration Fields
+
+```go
+// Resource contains configuration for a given resource.
+type Resource struct {
+    // Version specifies the version of the resource (e.g., "v1beta1")
+    Version string
+
+    // PreviousVersions is a list of previous API versions that should still
+    // be generated and served by the provider
+    PreviousVersions []string
+
+    // ... other fields
+}
+```
+
+### Example Configuration
+
+Here's how to configure a resource with multiple versions:
+
+```go
+func Configure(p *config.Provider) {
+    p.AddResourceConfigurator("azurerm_example_resource", func(r *config.Resource) {
+        // Current version
+        r.Version = "v1beta1"
+
+        // Previous versions to maintain
+        r.PreviousVersions = []string{"v1alpha2", "v1alpha1"}
+
+        // Additional configuration for conversions (see below)
+    })
+}
+```
+
+This configuration will:
+1. Generate CRDs for all three versions: `v1alpha1`, `v1alpha2`, and `v1beta1`
```

Comment:
This does not appear entirely accurate. Upjet only generates the latest (current) version. In there r.Version's value is `v1beta1`.